test/dsp.jl

```diff
@@ -72,13 +72,19 @@ end
 
         offset_arr = OffsetVector{Int}(undef, -1:2)
         offset_arr[:] = a
-        @test conv(offset_arr, 1:3) == OffsetVector(expectation, 0:5)
+        @test_throws ArgumentError conv(offset_arr, 1:3)
+        @test conv(offset_arr, OffsetArray(1:3)) == OffsetVector(expectation, 0:5)
         offset_arr_f = OffsetVector{Float64}(undef, -1:2)
         offset_arr_f[:] = fa
-        @test conv(offset_arr_f, 1:3) ≈ OffsetVector(fexp, 0:5)
+        @test_throws ArgumentError conv(offset_arr_f, 1:3)
+        @test conv(offset_arr_f, OffsetArray(1:3)) ≈ OffsetVector(fexp, 0:5)
         @test_throws ArgumentError conv!(zeros(6), offset_arr, 1:3) # output needs to be OA, too
         @test_throws ArgumentError conv!(OffsetVector{Int}(undef, 1:6), 1:4, 1:3) # output mustn't be OA
 
+        @test conv(fa, fill(true)) == conv(fill(true), fa) == fa
+        @test_broken conv(offset_arr_f, fill(true)) == conv(fill(true), offset_arr_f) == offset_arr_f
```

Comment:
As `fill(true)`  doesn't have any axes, this should arguably work. However, `conv` currently first brings all arguments to the same dimensionality, turning `fill(true)` into `[true]` here, and the axis compatibility check then says no. Fixing that is probably not important and certainly beyond the scope of this PR, but I thought leaving a `@test_broken` as a reminder might be in order.